### DISCUSSION TITLE
Is UnhandledExceptionDelegate handling resolvers?

### DISCUSSION THREAD

**Discussion by SebastianStehle** [graphql-dotnet/graphql-dotnet#3676] [Q&A]
Hi,

I have this error visitor that has been introduced some time. I am not sure, if I still need it, because there is also the UnhandledExceptionDelegate now.

If I understand it correctly this delegate also captures exceptions from resolvers, right?

**Comment by Shane32** [comment]
Yes, that is correct.  Although that is its principal purpose, it can also catch exceptions from document listeners or validation rules.  (Although the built-in rules should not throw any exceptions.)

Note that:
1. `ExecutionError` exceptions (or derived classes) are added to the errors collection of the returned `ExecutionResult` and are not handled by the `UnhandledExceptionDelegate`.
2. `OperationCancelledException`s are not handled by the `UnhandledExceptionDelegate`, but rather cause the exception to be rethrown back to the caller of `IDocumentExecuter.ExecuteAsync`.  So OCEs are not wrapped and returned within an `ExecutionResult` like other errors.
3. You can tailor the serialization of errors by providing a custom `IErrorInfoProvider` implementation or configuring options on the default implementation.

We also have extensive documentation on error handling here:
- https://graphql-dotnet.github.io/docs/getting-started/errors/
